File: Angular/src/app/model/ReportCaptcha.ts
export class ReportCaptcha {
    constructor(
        private id: string,
        private numberHits: number,
        private numberFaill: number,
        private lastData: string
    ) {}

    // Getters
    public getId(): string {
        return this.id;
    }

    public getNumberHits(): number {
        return this.numberHits;
    }

    public getNumberFaill(): number {
        return this.numberFaill;
    }

    public getLastData(): string {
        return this.lastData;
    }

    // Setters
    public setId(id: string): void {
        this.id = id;
    }

    public setNumberHits(numberHits: number): void {
        this.numberHits = numberHits;
    }

    public setNumberFaill(numberFaill: number): void {
        this.numberFaill = numberFaill;
    }

    public setLastData(lastData: string): void {
        this.lastData = lastData;
    }

    // Método para crear una instancia desde un objeto JSON
    public static fromJSON(data: any): ReportCaptcha {
        return new ReportCaptcha(
            data.id || '',
            data.numberHits || 0,
            data.numberFaill || 0,
            data.lastData || ''
        );
    }
}

File: CaptchaForge/src/main/java/com/cunoc/CaptchaForge/Model/WebIdentities/ReportCaptcha.java
package com.cunoc.CaptchaForge.Model.WebIdentities;

public class ReportCaptcha {
    
    private String id;
    private int numberHits;
    private int numberFaill;
    private String lastData;

    public ReportCaptcha(String id, int numberHits, int numberFaill, String lastData) {
        this.id = id;
        this.numberHits = numberHits;
        this.numberFaill = numberFaill;
        this.lastData = lastData;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public int getNumberHits() {
        return numberHits;
    }

    public void setNumberHits(int numberHits) {
        this.numberHits = numberHits;
    }

    public int getNumberFaill() {
        return numberFaill;
    }

    public void setNumberFaill(int numberFaill) {
        this.numberFaill = numberFaill;
    }

    public String getLastData() {
        return lastData;
    }

    public void setLastData(String lastData) {
        this.lastData = lastData;
    }

}
